File: rest/src/main/java/io/sparkled/viewmodel/playlist/search/PlaylistSearchViewModel.java
package io.sparkled.viewmodel.playlist.search;

import io.sparkled.viewmodel.ViewModel;

import java.util.Objects;

public class PlaylistSearchViewModel implements ViewModel {

    private Integer id;
    private String name;
    private Integer sequenceCount;
    private Integer durationSeconds;

    public Integer getId() {
        return id;
    }

    public PlaylistSearchViewModel setId(Integer id) {
        this.id = id;
        return this;
    }

    public String getName() {
        return name;
    }

    public PlaylistSearchViewModel setName(String name) {
        this.name = name;
        return this;
    }

    public Integer getSequenceCount() {
        return sequenceCount;
    }

    public PlaylistSearchViewModel setSequenceCount(Integer sequenceCount) {
        this.sequenceCount = sequenceCount;
        return this;
    }

    public Integer getDurationSeconds() {
        return durationSeconds;
    }

    public PlaylistSearchViewModel setDurationSeconds(Integer durationSeconds) {
        this.durationSeconds = durationSeconds;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        PlaylistSearchViewModel that = (PlaylistSearchViewModel) o;
        return Objects.equals(id, that.id) &&
                Objects.equals(name, that.name) &&
                Objects.equals(sequenceCount, that.sequenceCount) &&
                Objects.equals(durationSeconds, that.durationSeconds);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name, sequenceCount, durationSeconds);
    }

    @Override
    public String toString() {
        return "PlaylistSearchViewModel{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", sequenceCount=" + sequenceCount +
                ", durationSeconds=" + durationSeconds +
                '}';
    }
}

File: rest/src/main/kotlin/io/sparkled/viewmodel/playlist/search/PlaylistSearchViewModel.kt
package io.sparkled.viewmodel.playlist.search

import io.sparkled.viewmodel.ViewModel

class PlaylistSearchViewModel : ViewModel {

    private var id: Int? = null
    private var name: String? = null
    private var sequenceCount: Int? = null
    private var durationSeconds: Int? = null

    fun getId(): Int? {
        return id
    }

    fun setId(id: Int?): PlaylistSearchViewModel {
        this.id = id
        return this
    }

    fun getName(): String? {
        return name
    }

    fun setName(name: String?): PlaylistSearchViewModel {
        this.name = name
        return this
    }

    fun getSequenceCount(): Int? {
        return sequenceCount
    }

    fun setSequenceCount(sequenceCount: Int?): PlaylistSearchViewModel {
        this.sequenceCount = sequenceCount
        return this
    }

    fun getDurationSeconds(): Int? {
        return durationSeconds
    }

    fun setDurationSeconds(durationSeconds: Int?): PlaylistSearchViewModel {
        this.durationSeconds = durationSeconds
        return this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as PlaylistSearchViewModel

        if (id != other.id) return false
        if (name != other.name) return false
        if (sequenceCount != other.sequenceCount) return false
        if (durationSeconds != other.durationSeconds) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (name?.hashCode() ?: 0)
        result = 31 * result + (sequenceCount?.hashCode() ?: 0)
        result = 31 * result + (durationSeconds?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return "PlaylistSearchViewModel(id=$id, name=$name, sequenceCount=$sequenceCount, durationSeconds=$durationSeconds)"
    }
}
